### PR TITLE
Add cluster role binding in v4

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards [#2296](https://github.com/giantswarm/giantswarm/issues/2296)

When a guest cluster is created, kvm operator creates three deployments (worker, master, node-controller) that should have a service account and a role bound to be able to manage resources. Same as cluster role binding v3.

### CODE REVIEW

File: service/kvmconfig/v4/resource/clusterrolebinding/resource.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+func getClusterRoleBindingByName(list []*apiv1.ClusterRoleBinding, name string) (*apiv1.ClusterRoleBinding, error) {
+	for _, l := range list {
+		if l.Name == name {
+			return l, nil
+		}
+	}
+
+	return nil, microerror.Mask(notFoundError)
```
Comment (kopiczko) [inline service/kvmconfig/v4/resource/clusterrolebinding/resource.go]: I suggest adding some annotation what is not found.

File: service/kvmconfig/v4/resource/clusterrolebinding/resource.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "clusterrolebindingv4"
+)
+
+// Config represents the configuration used to create a new config map resource.
+type Config struct {
+	// Dependencies.
```
Comment (kopiczko) [inline service/kvmconfig/v4/resource/clusterrolebinding/resource.go]: Nitpick: we decided to remove `// Dependencies.`-like comments.

File: service/kvmconfig/v4/resource/clusterrolebinding/desired.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+		TypeMeta: apismetav1.TypeMeta{
+			Kind:       "ClusterRoleBinding",
+			APIVersion: apiv1.GroupName,
+		},
+		ObjectMeta: apismetav1.ObjectMeta{
+			Name: key.ClusterRoleBindingName(customObject),
+			Labels: map[string]string{
+				"app":         "kvm-operator",
+				"cluster-id":  key.ClusterID(customObject),
+				"customer-id": key.ClusterCustomer(customObject),
```
Comment (kopiczko) [inline service/kvmconfig/v4/resource/clusterrolebinding/desired.go]: giantswarm.io/cluster-id & giantswarm.io/customer-id
https://github.com/giantswarm/fmt/blob/master/kubernetes/annotations_and_labels.md

File: service/kvmconfig/v4/resource/clusterrolebinding/desired.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		TypeMeta: apismetav1.TypeMeta{
+			Kind:       "ClusterRoleBinding",
+			APIVersion: apiv1.GroupName,
+		},
+		ObjectMeta: apismetav1.ObjectMeta{
+			Name: key.ClusterRoleBindingPSPName(customObject),
+			Labels: map[string]string{
+				"app":         "kvm-operator",
+				"cluster-id":  key.ClusterID(customObject),
+				"customer-id": key.ClusterCustomer(customObject),
```
Comment (kopiczko) [inline service/kvmconfig/v4/resource/clusterrolebinding/desired.go]: Labels again.

**Comment by kopiczko** [comment]
I think it fails because of metrics registration. I'll fix that later today.

File: service/kvmconfig/v4/resource_set.go [hunk 5]
```diff
@@ -16,6 +16,7 @@ import (
 
 	"github.com/giantswarm/kvm-operator/service/kvmconfig/v3/cloudconfig"
 	"github.com/giantswarm/kvm-operator/service/kvmconfig/v3/key"
+	"github.com/giantswarm/kvm-operator/service/kvmconfig/v3/resource/clusterrolebinding"
```
Comment (kopiczko) [inline service/kvmconfig/v4/resource_set.go]: This should be v4 I'm about to merge https://github.com/giantswarm/kvm-operator/pull/380

Comment (kopiczko) [reply service/kvmconfig/v4/resource_set.go]: It says now you have to resolve conflicts.

**Comment by pipo02mix** [comment]
@kopiczko I tested it for v3 before, but If I want to double check with v4, I have to create a WIP version bundle, right? how do I proceed?

**Comment by kopiczko** [comment]
I added version bundle in this PR https://github.com/giantswarm/kvm-operator/pull/384 

Please after merging update the changelog.

PS I assigned you to the PR. Try to remember to self assign to PRs.

**Comment by pipo02mix** [comment]
About the changelog, these PRs has added any value for guest clusters. So I am not sure what write about in changelog

**Comment by kopiczko** [comment]
Re changelog: nothing was fixed?

Please also remove `copy v3` from the commit message for my OCD :)

**Comment by pipo02mix** [comment]
Not for guest clusters. I added cluster role bindings that have an impact on the RBAC for the host cluster. I see now has come a change for kubernetes.

I have tested and it works fine.

**Comment by pipo02mix** [comment]
PTAL and I merge

**Pull request merged**